File: Backend/Models/InterviewChain.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    public class InterviewChain
    {
        [Key]
        public int InterviewChainID { get; set; }

        [ForeignKey("ParentInterviewChain")]
        public int? ParentInterviewChainID { get; set; } // Links to previous chain for same EndClientName
        public InterviewChain? ParentInterviewChain { get; set; }

        [ForeignKey("Client")]
        public int? ClientID { get; set; }
        public Client? Client { get; set; }

        public string? EndClientName { get; set; } // From old interviews table

        [StringLength(100)]
        public string? Position { get; set; } // Job position

        [StringLength(50)]
        public string? ChainStatus { get; set; } // "Active", "Successful", "Unsuccessful"

        public int Rounds { get; set; } // Number of interviews in the chain

        [Required]
        public DateTime InterviewEntryDate { get; set; } // Date the chain was entered

        [ForeignKey("Recruiter")]
        public int? RecruiterID { get; set; }
        public Employee? Recruiter { get; set; }

        public DateTime? InterviewDate { get; set; }
        [StringLength(50)]
        public string? InterviewStartTime { get; set; }
        [StringLength(50)]
        public string? InterviewEndTime { get; set; }
        [StringLength(100)]
        public string? InterviewMethod { get; set; }
        [StringLength(100)]
        public string? InterviewType { get; set; }
        [StringLength(50)]
        public string? InterviewStatus { get; set; } // "Scheduled", "In Progress", "Completed"
        [StringLength(50)]
        public string? InterviewOutcome { get; set; } // "Next", "Offer", "Rejected"
        public string? InterviewSupport { get; set; }
        public string? InterviewFeedback { get; set; }
        public string? Comments { get; set; }

        [Required]
        public DateTime CreatedTS { get; set; } = DateTime.UtcNow;
        [Required]
        public DateTime UpdatedTS { get; set; } = DateTime.UtcNow;

        [StringLength(100)]
        public string? CreatedBy { get; set; }
        [StringLength(100)]
        public string? UpdatedBy { get; set; }

        // Navigation property for interviews in the chain
        public virtual ICollection<InterviewChain> ChildInterviews { get; set; } = new List<InterviewChain>();
    }
}

File: ReactApp/app/types/interviewChain/interviewChain.ts
// types.ts

export interface Interview {
  InterviewChainID: number;
  ParentInterviewChainID: number | null | undefined;
  ClientID: number | null;
  EndClientName: string;
  Position: string;
  ChainStatus: string;
  Rounds: number;
  InterviewEntryDate: Date | null;
  RecruiterID: number | null;
  InterviewDate: Date | null;
  InterviewStartTime: string | null;
  InterviewEndTime: string | null;
  InterviewMethod: string | null;
  InterviewType: string | null;
  InterviewStatus: string;
  InterviewOutcome: string | null;
  InterviewSupport: string | null;
  InterviewFeedback: string | null;
  Comments: string | null;
  CreatedTS: Date | null;
  UpdatedTS: Date | null;
  CreatedBy: string | null;
  UpdatedBy: string | null;
}

export interface InterviewChain {
  id: string;
  endClientName: string;
  clientName: string;
  recruiterName: string;
  position: string;
  status: string;
  interviews: Interview[];
  rounds: number;
  createdAt: string;
  updatedAt: string;
  latestInterview: Interview | null;
  latestInterviewDate?: string | null;
  latestInterviewStatus?: string | null;
  latestInterviewType?: string | null;
}

export interface InterviewChainStats {
  totalChains: number;
  activeChains: number;
  successfulChains: number;
  unsuccessfulChains: number;
  statusBreakdown: { status: string; count: number }[];
  monthlyActivity: {
    month: string;
    active: number;
    successful: number;
    unsuccessful: number;
  }[];
  averageRounds: number;
  offerRate: number;
  topClients: { name: string; count: number }[];
}

export interface InterviewChainList {
  interviewChainID: number;
  clientID?: number | null;
  clientName?: string | null;
  endClientName?: string | null;
  position?: string | null;
  chainStatus?: string | null;
  rounds: number;
  interviewEntryDate: string;
  recruiterID?: number | null;
  recruiterName?: string | null;
  latestInterviewDate?: string | null;
  latestInterviewStatus?: string | null;
  latestInterviewType?: string | null;
}

export interface InterviewChainDetail {
  interviewChainID: number;
  parentInterviewChainID?: number | null;
  clientID?: number | null;
  clientName?: string | null;
  endClientName?: string | null;
  position?: string | null;
  chainStatus?: string | null;
  rounds: number;
  interviewEntryDate: string;
  recruiterID?: number | null;
  recruiterName?: string | null;
  interviews: { $values: any[] };
}

export interface InterviewChainCreate {
  clientID?: number | null;
  endClientName?: string | null;
  position?: string | null;
  recruiterID?: number | null;
  interviewEntryDate: string;
  interviewDate?: string | null;
  interviewStartTime?: string | null;
  interviewEndTime?: string | null;
  interviewMethod?: string | null;
  interviewType?: string | null;
  interviewStatus?: string;
  interviewSupport?: string | null; // Added
  comments?: string | null;
}

export interface InterviewChainUpdate {
  interviewChainID: number;
  chainStatus?: string | null;
  interviewDate?: string | null;
  interviewStartTime?: string | null;
  interviewEndTime?: string | null;
  interviewMethod?: string | null;
  interviewType?: string | null;
  interviewStatus?: string | null;
  interviewOutcome?: string | null;
  interviewSupport?: string | null; // Added
  interviewFeedback?: string | null;
  comments?: string | null;
}

export interface InterviewChainEnd {
  interviewChainID: number;
  interviewOutcome?: string | null;
  interviewFeedback?: string | null;
  comments?: string | null;
}

export interface InterviewChainAdd {
  interviewChainID: number;
  parentInterviewChainID?: number;
  interviewDate?: string | null;
  interviewStartTime?: string | null;
  interviewEndTime?: string | null;
  interviewMethod?: string | null;
  interviewType?: string | null;
  interviewStatus?: string;
  interviewSupport?: string | null; // Added
  comments?: string | null;
}

export interface InterviewChainCreateResponse {
  interviewChainID: number;
}

export interface ChainExplorationProps {
  chain: InterviewChain;
  open: boolean;
  onClose: () => void;
  onEndInterview: (
    chain: InterviewChain,
    isEditing: boolean,
    interview?: Interview
  ) => void;
  onAddNewInterview: (
    chain: InterviewChain,
    newInterview?: Partial<Interview>
  ) => void;
  onUpdateChainStatus: (chainId: string, newStatus: string) => void;
  onEditInterview: (interview: Interview) => void;
}

export interface EndInterviewDialogProps {
  chain: InterviewChain;
  open: boolean;
  onClose: () => void;
  onSubmit: (
    chainId: string,
    outcome: "Next" | "Rejected" | "Offer" | "AddNew" | "Edit",
    newInterview?: Partial<Interview> & {
      clientName?: string;
      position?: string;
      recruiterName?: string;
    }
  ) => void;
  onOpenAddInterview: () => void;
  selectedInterview?: Interview;
  isSubmitting?: boolean;
}

export interface AddInterviewDialogProps {
  chain: InterviewChain;
  open: boolean;
  onClose: () => void;
  onSubmit: (
    chainId: string,
    outcome: "AddNew",
    newInterview?: Partial<Interview> & {
      clientName?: string;
      position?: string;
      recruiterName?: string;
    }
  ) => void;
  selectedInterview?: Interview;
  isSubmitting?: boolean;
}

export interface EditInterviewDialogProps {
  chain: InterviewChain;
  open: boolean;
  onClose: () => void;
  onSubmit: (
    chainId: string,
    outcome: "Edit",
    newInterview?: Partial<Interview> & {
      clientName?: string;
      position?: string;
      recruiterName?: string;
    }
  ) => void;
  interviewToEdit: Interview;
  isSubmitting?: boolean;
}
